### ISSUE TITLE
Bug: Loads `.eslintrc.js` even when `useEslintrc: false` and `overrideConfig: {...}` is set 

### ISSUE THREAD

**Issue by sfarthin** [eslint/eslint#16723]
### Environment

Node version: 16.15.1
npm version: 8.11.0
Local ESLint version: v8.23.0
Global ESLint version: none
Operating System: MacOS

I want to have a script to "remove unused imports" inside a larger project. Specifically, as part of a codemod.

This larger project has an `.eslintrc.js` in the root that has a series of hacks to get it to work as we want it. This works _fine_, but it breaks my script. If I delete `eslintrc.js` and run this script it works great, but I don't want to remove `.eslintrc.js`! I see various options for overrides or to ignore this `eslintrc.js`, but they must still be loading this config. If it exists, my script fails.

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
// It's a big one, but it shouldn't matter, I don't want to load any config file.
```
</details>

```ts

```ts
  // example script

  const eslint = new ESLint({
    fix: true,
    useEslintrc: false, // <-- I thought this should stop eslint from loading .eslintrc, doesn't work
    cwd: __dirname, // <-- another hack to try to stop eslint from loading .eslintrc, didn't work
    overrideConfig: { // <-- when we override the config, we shouldn't need to load the core config, right? didn't work
      parser: '@typescript-eslint/parser',
      settings: {
        'import/parsers': {
          '@typescript-eslint/parser': ['.ts', '.tsx', '.d.ts'],
        },
        'import/resolver': {
          typescript: {
            project: './tsconfig.json',
          },
        },
      },
      extends: [],
      plugins: ['unused-imports'],
      rules: {
        // These are the only rules applicable to my script.
        'unused-imports/no-unused-imports': 'error', 
        'unused-imports/no-unused-vars': 'error',
      },
    },
  });

  const results = await eslint.lintText(input, { filePath });

  const output = results[0].output;

```

### What did you expect to happen?

Do no load `eslintrc.js` if `useEslintrc: false`. I am open to any hack to get this to work.

### What actually happened?

```
Failed to load plugin 'jsx-a11y' declared in '.eslintrc.js » /Users/SFarthing/truebill-native/.tooling/eslint-config-airbnb.js » /Users/SFarthing/truebill-native/node_modules/eslint-config-airbnb/rules/react-a11y.js': Cannot read properties of undefined (reading 'document')
```

I continually get this error.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by sfarthin** [comment]
ok weird, I finally was able to get this to work like this: 

```const eslint = new ESLint({
  fix: true,
  useEslintrc: false,
  overrideConfig: {
    parser: '@typescript-eslint/parser',
    settings: {
      'import/parsers': {
        '@typescript-eslint/parser': ['.ts', '.tsx', '.d.ts'],
      },
      'import/resolver': {
        typescript: {
          project: './tsconfig.json',
        },
      },
    },
    extends: [],
    // This plugin will automatically remove unused imports
    plugins: ['unused-imports'],
    rules: {
      'unused-imports/no-unused-imports': 'error',
      'unused-imports/no-unused-vars': 'error',
    },
  },
});```

dunno why it wasnt working before
